### ISSUE TITLE
PSR12.Operators.OperatorSpacing and PHP 8.0 union types

### ISSUE THREAD

**Issue by rask** [squizlabs/PHP_CodeSniffer#3151]
**Describe the bug**
When using the PSR-12 implementation that ships with PHP_CodeSniffer, I cannot use union types as described in RFCs and other official or semi-official documentation.

**Code sample**
```php
function foo(int|float $val) : void
{
     //
}
```

**Custom ruleset**
```xml
<?xml version="1.0"?>
<ruleset name="My Custom Standard">
    <rule ref="PSR12" />
</ruleset>
```

**To reproduce**
1. Lint the file while running PHP 8.0 so you can use union types
2. Get greeted with `PSR12.Operators.OperatorSpacing` errors (either `NoSpaceBefore` or `NoSpaceAfter`):
```
ERROR | [x] Expected at least 1 space after "|"; 0 found
```

**Expected behavior**
I would expect to be able to use PSR-12 operator spacing lints without it complaining about union type delimiters, as those to my understanding are not actually operators.

**Versions (please complete the following information):**
 - OS: Linux Ubuntu
 - PHP: 8.0 RC 1
 - PHPCS: Latest 3.5 stable
 - Standard: PSR12

**Additional context**
N/A

**Comment by jrfnl** [comment]
Union types aren't supported yet in PHP_CodeSniffer.

A PR to fix this is open and expected to be merged in PHPCS 3.6.0: #3032

**Comment by jrfnl** [comment]
As PR #3032 has been merged, this issue can be closed as fixed.

**Comment by gsherwood** [comment]
Can confirm fixed

**Comment by fr0N73ND3r** [comment]
Hello. This still throws an error in the "catch" section

```
    try {
    } catch (\Throwable|\Exception $e) {
    }
```
```
vendor/bin/phpcs --standard=PSR12 src/
```

ERROR | [x] Expected at least 1 space before "|"; 0 found

**Comment by jrfnl** [comment]
@fr0N73ND3r That's because those are not considered union types, but a multi-catch condition. May look the same, but it is technically something different.

I'd suggest bringing this up with PSR PER to add explicit rules about this.
